dataloader.py

```diff
@@ -0,0 +1,113 @@
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Tuple
+
+import cv2
+import torch
+from skimage import io
+from torch import Tensor
+from torch.utils.data import Dataset
+from torchvision.transforms import RandomResizedCrop
+
+from constants import NO_VAL, TEST_DIR, TRAIN_DIR, VAL_DIR
+
+# Make a hashmap for both datasets
+
+
+class ChestXrayDataset(Dataset):
+    def __init__(
+        self,
+        phase: str,
+        crop: bool = False,
+        patch_size: int = 64,
+        transfom: Callable = None,
+    ) -> None:
+        super().__init__()
+        self.phase = phase
+        if self.phase == "train":
+            self.img_paths = list(TRAIN_DIR.rglob("*.jpeg"))
```

Comment:
Can you be certain that the behaviour of `Path.rglob` is consistent? E.g. is is possible the ordering of paths returned is different each time, because `rglob` uses a hash of some kind internally? Remove doubt by making sure:

```suggestion
            self.img_paths = sorted(TRAIN_DIR.rglob("*.jpeg"))
```

Reply:
Done